client/client.go

```diff
@@ -197,6 +197,7 @@ func (c *clientConfig) tryPopulateInfo(clients ...Client) (err error) {
 			if err == nil {
 				return
 			}
+
 			if ctx.Err() != nil {
```

Comment:
not part of your code change, but this looks a little suspicious - checking an err == nil and then checking an err a different way.  Looks a bit like the second check might not happen or be effective!~

Reply:
But can ignore for this PR, just stream of consciousness reviewing

Reply:
It's actually correct, the `ctx.Err()` check is there to check if the stream wasn't cancelled or whatsoever, it's checking the context is still valid basically. Whereas the `err == nil` is checking the `Info` function worked as expected.

Reply:
if `err != nil` and `ctx.Err() == nil` for all clients, it will return without returning an err to the caller though, and the caller does: `if err := cfg.tryPopulateInfo(cfg.clients...); err != nil` they could expect `chainInfo` to be set.

Reply:
It's a named return value: `(err error)`, so it will return the last error it got, or will return the context error.